containers/init/build/csharp/Dockerfile

```diff
@@ -0,0 +1,45 @@
+# Copyright 2020 gRPC authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+FROM mcr.microsoft.com/dotnet/sdk:2.1
+
+RUN mkdir -p /src/workspace
+WORKDIR /src/workspace
+
+RUN apt-get update && apt-get install -y \
+  build-essential \
+  git && \
+  apt-get clean
+
+# The prerequisites installed below are heavily inspired by the C# testing image from grpc/grpc
+# https://github.com/grpc/grpc/blob/master/tools/dockerfile/test/csharp_stretch_x64/Dockerfile
+
+# cmake >=3.6 needed to build grpc_csharp_ext
+RUN apt-get update && apt-get install -y cmake && apt-get clean
+
+# Python is needed to run the run_tests.py script (we use it to build C#)
+RUN apt-get update && apt-get install -y \
+  python-dev \
+  python-setuptools \
+  python-yaml && \
+  apt-get clean
+
+RUN curl https://bootstrap.pypa.io/get-pip.py | python2.7
```

Comment:
If we stick with run_tests.py... I thought we decided not to use python2.7? If you are having trouble getting python3 to work with grpc tools, ping me... I've been through the problem before. 

While we do not have to use run_tests.py, it is the easiest according to https://github.com/grpc/grpc/tree/master/src/csharp#build-from-source. There are other ways which would remove the need for python. But looking at the code for run_tests.py, C# looks a little complicated https://github.com/grpc/grpc/blob/9293ce3e81d5e6a3fc07e3263210f159e3b5b9cc/tools/run_tests/run_tests.py#L903.

Reply:
I haven't update this yet, I would eliminate the run_test.py as we discussed in the meeting, and therefor we would no longer need python for anything. :)